### ISSUE TITLE
Unsaved changes warning does not work for sub forms phase 2

### ISSUE THREAD

**Issue by bernardsilenou** [hzi-braunschweig/SORMAS-Project#3238]
### Feature Description
This is a  refinement of issue #3042
### Problem Description
 - Unsaved changes is either does not get triggered or trigger after navigation when creating or editing other forms linked to the current form. Consider the case form for example, we can find associated forms like samples, task, events, etc. Opening any of these forms should trigger unsaved change warning.
- System allows you to navigate and fill the form for new sample and test results (just for example), when you then save this from for the new sample, system informs you of the unsaved  changes on the case form you left long ago. 

### Proposed Change
Clicking on the new + or edit  icons should trigger "unsaved changes" warning

For a case, I suggest the warning is not needed when creating a new location but needed for:
- new  samples, contacts; events, 
- new test result of a sample
- editing test result of a sample
- Events: edit task, new task, edit action, new actions
- Cases: New task, edit task, new sample, link to event
- Any action that takes the user out of the current form for a while

**Comment by kwa20** [comment]
@bernardsilenou Do you think this might also be associated with the prompt being triggered when (seemingly) no editations have been made? People keep reporting it popping up although no changes have been made.

**Comment by bernardsilenou** [comment]
> @bernardsilenou Do you think this might also be associated with the prompt being triggered when (seemingly) no editations have been made? People keep reporting it popping up although no changes have been made.

@kwa20 I do not think this is related although both issues fall in the same group. The issue you mentioned was resolved in 1,50.

**Comment by kwa20** [comment]
@bernardsilenou People still reported it happening in the test protocols for 1.50, I'll try reproducing it again and try to pin point down the navigations in which this seems to occur.

**Comment by Candice-Louw** [comment]
@bernardsilenou @kwa20 - This has been newly reported after implementing the GIS servers and geo button in Switzerland, 2 or 3 days ago -
"Since the update to version 1.51.1, the prompt "Confirm changes" appears if you have not changed anything or directly after saving manually, if you want to exit the 'Case> Person case' tab and GPS coordinates are stored.
The same happens when you exit 'Contact> Contact person' (if GPS coordinates are stored).
Desired: the message should only appear if something has really been changed."

**Comment by kwa20** [comment]
@Candice-Louw This seems to be part of issue #3014
